File: customer-service/customer-ui/src/service/customerService.ts
import ApiRoutes from "../core/ApiRoutes";
import http from "../core/http-common";
import CustomerCommand from "../type/customer";
import CustomerResponse from "../type/customer";
import DriverResponse from "../type/driver";
import CustomerRequestDriver from "../type/request";


const findAllByDeletedFalse = () => {
    return http.get<Array<CustomerResponse>>(ApiRoutes.about);
}
const create = (payload: CustomerCommand) => {
    return http.post<CustomerCommand>(ApiRoutes.about, payload);
}
const findById = (id: string) => {
    return http.get<CustomerResponse>(ApiRoutes.about +`/${id}`);
}
const updateInfo = (id: any, payload: CustomerCommand) => {
    return http.put(ApiRoutes.about +`/${id}`, payload);
}
const deleteById = (id: any) => {
    return http.delete(ApiRoutes.about +`/${id}`);
}
const getDriversAvailable = () => {
    return http.get<DriverResponse>(ApiRoutes.availableDrivers);
}
const sendRequestToDriver = (request: CustomerRequestDriver) => {
    return http.post<string>(ApiRoutes.request);
}

const customerService = {
    create,
    findAllByDeletedFalse,
    findById,
    updateInfo,
    deleteById,
    getDriversAvailable,
    sendRequestToDriver
}
export default customerService;

File: customer-service/src/main/java/com/nas/customer/service/service/customer/CustomerService.java
package com.nas.customer.service.service.customer;

import com.nas.customer.service.command.CustomerCommand;
import com.nas.customer.service.command.CustomerInfoUpdateCmd;
import com.nas.customer.service.command.CustomerRequestDriver;
import com.nas.customer.service.command.RatingCommand;
import com.nas.customer.service.criteria.CustomerCriteria;
import com.nas.customer.service.model.Customer;
import com.nas.customer.service.model.Driver;
import com.nas.customer.service.payload.CustomerDetails;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Set;

public interface CustomerService {
    Customer create(final CustomerCommand customerCommand);
    Page<Customer> findAllByDeletedFalse(Pageable pageable);
    Customer findById(String customerId);
    void updateInfo(final CustomerInfoUpdateCmd customerCommand, String customerId);
    Set<Driver> getDriversAvailable();
    void sendRequestDriver(CustomerRequestDriver requestDriver);
    String sendRating(final RatingCommand ratingCommand);
    Page<Customer> getAllByCriteria(Pageable pageable, CustomerCriteria customerCriteria);
    CustomerDetails findCustomerDetailsById(String customerId);
}